hrjobcontract/js/src/job-contract/controllers/contract.js

```diff
@@ -77,7 +77,7 @@ define([
               * @param {string || date} newEndDate the date specified by the user
             */
             function updateContractList(newEndDate){
-              var isCurrentContract = moment().diff(newEndDate, "day") <= 0;
+              var isCurrentContract = !newEndDate ? true : (moment().diff(newEndDate, "day") <= 0);
```

Comment:
1. check if !newEndDate will pass through for empty strings or empty objects, what are the various expected values of newEndDate?
2. also check if test cases are updated for this